govuk-one-login/authentication-api
# Aut 3613/return reauth lock info from start handler

## What

Returns reauth lock information from the start handler if the feature flag to use the new authentication attempts service is on.

Also removes the existing deletion of the counts, since we are going to retain these in order to return blocked information.

This is currently unused on the frontend, but a subsequent PR will respond on the frontend in the case of a reauth journey where someone is blocked.

## How to review

1. Code Review commit by commit

How I've tested this (you can choose to do the same or trust me):
1. Deploy to a test env, with the relevant reauth feature flags on
1. Open two different browsers. Log in in both
1. Start a reauth journey in one. Enter the wrong email or the wrong password to the point where you're logged out
1. Start a reauth journey in your other browser, just to the point where you hit the enter email page. You won't be blocked, but check the logs for the start handler and see that for this last call, isBlockedForReauth is true

Note! You need to do this all quite quickly to make sure sessions and ttls don't expire, worth getting it well set up in advance. 

1. Then check that for a journey with the feature flags off, nothing mad happens.
